scripts/compress_images_test.py

```diff
@@ -0,0 +1,261 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for compress_images.py."""
+
+from __future__ import annotations
+
+import os
+import pathlib
+import shutil
+import tempfile
+import unittest
+from unittest import mock
+
+from scripts import compress_images
+
+from PIL import Image
+from typing import List, TypedDict
+
+
+class CompressedImageInfo(TypedDict):
+    """Type definition for compressed image information."""
+
+    path: pathlib.Path
+    original_size: int
+    new_size: int
+
+
+class TestImageCompressor(unittest.TestCase):
+    """Test the ImageCompressor class."""
+
+    def setUp(self) -> None:
+        """Set up test environment before each test."""
+        self.temp_dir = tempfile.mkdtemp()
+        self.output_dir = os.path.join(self.temp_dir, 'compressed_images')
+
+        self.png_path = os.path.join(self.temp_dir, 'test_image.png')
+        self.jpg_path = os.path.join(self.temp_dir, 'test_image.jpg')
+        self.webp_path = os.path.join(self.temp_dir, 'test_image.webp')
+
+        self._create_test_image(self.png_path, 'PNG')
+        self._create_test_image(self.jpg_path, 'JPEG')
+        self._create_test_image(self.webp_path, 'WEBP')
+
+    def tearDown(self) -> None:
+        """Clean up test environment after each test."""
+        shutil.rmtree(self.temp_dir)
+
+    def _create_test_image(self, path: str, Img_format: str) -> None:
+        """Create a test image for compression testing."""
+        img = Image.new('RGB', (100, 100), color='red')
+        img.save(path, format=Img_format)
+
+    def test_get_compression_type_zip_extension(self) -> None:
+        """Test get_compression_type returns 'Zip' for PNG."""
+        compressor = compress_images.ImageCompressor(self.temp_dir)
+        compression_type = compressor.get_compression_type('.png')
+        self.assertEqual(compression_type, 'Zip')
+
+    def test_get_compression_type_lzw_extension(self) -> None:
+        """Test get_compression_type returns 'LZW' for JPG."""
+        compressor = compress_images.ImageCompressor(self.temp_dir)
+        compression_type = compressor.get_compression_type('.jpg')
+        self.assertEqual(compression_type, 'LZW')
+
+    @mock.patch('subprocess.run')
+    def test_compress_images(self, mock_subprocess_run: mock.MagicMock) -> None:
+        """Test image compression process."""
+        self.assertFalse(os.path.exists(self.output_dir))
+        result_images: List[CompressedImageInfo] = [
+            {
+                'path': pathlib.Path(self.png_path),
+                'original_size': 1000,
+                'new_size': 500
+            }
+        ]
+        compressor = compress_images.ImageCompressor(
+            self.temp_dir,
+            output_dir=self.output_dir
+        )
+        compressor.compress_images(result_images)
+        self.assertTrue(os.path.exists(self.output_dir))
```

Comment:
can you strengthen this assertion? Right now we're just asserting that the output path exists, but not that it actually contains compressed images. Maybe save in your test assets folder a copy of what the compressed image should look like (assuming the compression is deterministic) and compare against that? Alternatively, if compression isn't deterministic, you could check file sizes